prov/efa/src/rdm/efa_rdm_cq.c

```diff
@@ -300,15 +301,20 @@ void efa_rdm_cq_poll_ibv_cq(ssize_t cqe_to_process, struct efa_ibv_cq *ibv_cq)
 	struct efa_rdm_ep *ep = NULL;
 	struct fi_cq_err_entry err_entry;
 	struct efa_rdm_cq *efa_rdm_cq;
+	struct efa_domain *efa_domain;
+	struct efa_qp *qp;
+
+	efa_rdm_cq = container_of(ibv_cq, struct efa_rdm_cq, ibv_cq);
+	efa_domain = container_of(efa_rdm_cq->util_cq.domain, struct efa_domain, util_domain);
 
 	/* Call ibv_start_poll only once */
 	err = ibv_start_poll(ibv_cq->ibv_cq_ex, &poll_cq_attr);
 	should_end_poll = !err;
 
 	while (!err) {
 		pkt_entry = (void *)(uintptr_t)ibv_cq->ibv_cq_ex->wr_id;
-		ep = pkt_entry->ep;
-		assert(ep);
+		qp = efa_domain->qp_table[ibv_wc_read_qp_num(ibv_cq->ibv_cq_ex) & efa_domain->qp_table_sz_m1];
```

Comment:
Could you remind me why we need to `... & qp_table_sz_m1`?

Reply:
I believe it is just for safe, we have the qp table of size `qp_table_size = roundup_power_of_two(efa_domain->device->ibv_attr.max_qp);`, and qp_num shouldn't exceeds `qp_num < ibv_attr.max_qp < qp_table_size`.